src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+            ImmutableArray<ExpressionSyntax> expandedArguments,
+            InterpolatedStringExpressionSyntax interpolatedString)
+        {
+            return interpolatedString.ReplaceNodes(interpolatedString.Contents, (oldNode, newNode) =>
+            {
+                var node = newNode as InterpolationSyntax;
+                if (node == null)
+                {
+                    return newNode;
+                }
+
+                var literalExpression = node.Expression as LiteralExpressionSyntax;
+                if (literalExpression != null && literalExpression.IsKind(SyntaxKind.NumericLiteralExpression))
+                {
+                    var index = (int)literalExpression.Token.Value;
+                    if (index >= 0 && index < expandedArguments.Length)
+                    {
+                        return node.WithExpression(FixTrivia(expandedArguments[index]));
+                    }
+                }
+
+                return newNode;
+            });
+        }
+
+        /// <summary>
+        /// Since C# interpolations cannot be on more than one line, we need to remove newlines if possible.
+        /// </summary>
+        public static ExpressionSyntax FixTrivia(ExpressionSyntax node) =>
+            node.ReplaceTokens(node.DescendantTokens(descendIntoTrivia: true), RemoveTriviaForTokens);
+
+        private static SyntaxToken RemoveTriviaForTokens(SyntaxToken originalToken, SyntaxToken rewrittenToken) =>
+            rewrittenToken
+                .WithLeadingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.LeadingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia))
+                .WithTrailingTrivia(
+                    ReplaceSyntaxKindsWithElasticSpace(rewrittenToken.TrailingTrivia, SyntaxKind.WhitespaceTrivia, SyntaxKind.EndOfLineTrivia));
+
+        private static SyntaxTriviaList ReplaceSyntaxKindsWithElasticSpace(SyntaxTriviaList trivialList, params SyntaxKind[] kinds) =>
+            trivialList.Select(x => kinds.Any(y => x.IsKind(y)) ? SyntaxFactory.ElasticSpace : x).ToSyntaxTriviaList();
+
+        private static ExpressionSyntax CastAndParenthesize(ExpressionSyntax expression, SemanticModel semanticModel) =>
+            Parenthesize(Cast(expression, semanticModel.GetTypeInfo(expression).ConvertedType));
+
+        private static ExpressionSyntax Cast(ExpressionSyntax expression, ITypeSymbol targetType)
+        {
+            if (targetType == null)
+            {
+                return expression;
+            }
+
+            var type = ParseTypeName(targetType.ToDisplayString());
+
+            return CastExpression(type, Parenthesize(expression))
+                .WithAdditionalAnnotations(Simplifier.Annotation);
+        }
+
+        private static ExpressionSyntax Parenthesize(ExpressionSyntax expression) =>
+            expression.IsKind(SyntaxKind.ParenthesizedExpression)
```

Comment:
Good catch! I suspect this, and possibly other duplicated helpers, exists because the original extension in C# Essentials didn't have access to internal helpers. These should be cleaned up.